sdk/storage/Azure.Storage.Files.Shares/src/ShareClient.cs

```diff
@@ -220,11 +230,48 @@ public ShareClient(Uri shareUri, StorageSharedKeyCredential credential, ShareCli
         /// <remarks>
         /// This constructor should only be used when shared access signature needs to be updated during lifespan of this client.
         /// </remarks>
-        public ShareClient(Uri shareUri, AzureSasCredential credential, ShareClientOptions options = default)
+        public ShareClient(
+            Uri shareUri,
+            AzureSasCredential credential,
+            ShareClientOptions options = default)
             : this(shareUri, credential.AsPolicy<ShareUriBuilder>(shareUri), options, sasCredential: credential)
         {
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ShareClient"/>
```

Comment:
TODO need to specify that OAuth only works with Create and Get Permission on the Share level.